### DISCUSSION TITLE
Number of matches in code search bar area

### DISCUSSION THREAD

**Discussion by OneTouchConor** [josdejong/svelte-jsoneditor#308] [Ideas]
I was wondering if it would be possible to add the number of matches found in the search to the Code view search area similar to that in the Tree view search area?
![Tree_SearchBar](https://github.com/josdejong/svelte-jsoneditor/assets/144229812/197ff194-9b9c-4233-b83f-0eb774b940d9)
![Code_Searchbar](https://github.com/josdejong/svelte-jsoneditor/assets/144229812/38e15bd3-600e-422a-b8f4-57e926af0fca)

I've also had a request from some users asking if there could be some way to add custom labels to the search. For example we have a JSON object with an array of 27 ContentBlocks and there may be matches for the search term in blocks 1, 4 and 17. They would like some way, at a glance, to be able to tell which block they are in for example hovering over the highlighted text may display the part of the object they are in. They are using the Code view to have access to the "match case" checkbox for search and replace so cannot use the Tree view to be able to see which index of the array they are on.

I was asked to check if these were possible and hopefully will be able to let the users know one way or the other, thanks!

**Comment by josdejong** [comment]
Thanks for your input, having the search result count would be very useful indeed. There is already an open issue about this in #63.

Can you explain your other ideas in more detail in a separate discussion to keep it focused? (I don't fully understand the idea of custom labels in search, but implementing "match case" in the searchbox of tree mode sounds very useful).
